{
  "cve": "CVE-2018-10875",
  "mitre": {
    "cpes": [],
    "created": "2018-07-13T22:00:00+00:00",
    "description": "A flaw was found in ansible. ansible.cfg is read from the current working directory which can be altered to make it point to a plugin or a module path under the control of an attacker, thus allowing the attacker to execute arbitrary code.",
    "metrics": {
      "cvssV2_0": {},
      "cvssV3_0": {
        "score": 7.8,
        "vector": "CVSS:3.0/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H"
      },
      "cvssV3_1": {},
      "cvssV4_0": {}
    },
    "mitre_repo_path": "cves/2018/10xxx/CVE-2018-10875.json",
    "references": [
      "http://lists.opensuse.org/opensuse-security-announce/2019-04/msg00021.html",
      "http://www.securitytracker.com/id/1041396",
      "https://access.redhat.com/errata/RHBA-2018:3788",
      "https://access.redhat.com/errata/RHSA-2018:2150",
      "https://access.redhat.com/errata/RHSA-2018:2151",
      "https://access.redhat.com/errata/RHSA-2018:2152",
      "https://access.redhat.com/errata/RHSA-2018:2166",
      "https://access.redhat.com/errata/RHSA-2018:2321",
      "https://access.redhat.com/errata/RHSA-2018:2585",
      "https://access.redhat.com/errata/RHSA-2019:0054",
      "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-10875",
      "https://lists.debian.org/debian-lts-announce/2019/09/msg00016.html",
      "https://usn.ubuntu.com/4072-1/",
      "https://www.debian.org/security/2019/dsa-4396"
    ],
    "title": null,
    "updated": "2024-08-05T07:46:47.518000+00:00",
    "vendors": [],
    "weaknesses": [
      "CWE-426"
    ]
  },
  "nvd": {
    "cpes": [
      "cpe:2.3:a:redhat:ansible_engine:2.0:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:ansible_engine:2.4:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:ansible_engine:2.5:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:ansible_engine:2.6:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:ceph_storage:2.0:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:ceph_storage:3.0:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:gluster_storage:3.0.0:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:openshift:3.0:*:*:*:enterprise:*:*:*",
      "cpe:2.3:a:redhat:openstack:10:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:openstack:12:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:openstack:13:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:virtualization:4.0:*:*:*:*:*:*:*",
      "cpe:2.3:a:redhat:virtualization_host:4.0:*:*:*:*:*:*:*",
      "cpe:2.3:a:suse:package_hub:-:*:*:*:*:*:*:*",
      "cpe:2.3:o:canonical:ubuntu_linux:16.04:*:*:*:lts:*:*:*",
      "cpe:2.3:o:canonical:ubuntu_linux:18.04:*:*:*:lts:*:*:*",
      "cpe:2.3:o:canonical:ubuntu_linux:19.04:*:*:*:*:*:*:*",
      "cpe:2.3:o:debian:debian_linux:8.0:*:*:*:*:*:*:*",
      "cpe:2.3:o:debian:debian_linux:9.0:*:*:*:*:*:*:*",
      "cpe:2.3:o:suse:suse_linux_enterprise_server:12:*:*:*:*:*:*:*"
    ],
    "created": "2018-07-13T22:29:00.220000+00:00",
    "description": "A flaw was found in ansible. ansible.cfg is read from the current working directory which can be altered to make it point to a plugin or a module path under the control of an attacker, thus allowing the attacker to execute arbitrary code.",
    "metrics": {
      "cvssV2_0": {
        "score": 4.6,
        "vector": "AV:L/AC:L/Au:N/C:P/I:P/A:P"
      },
      "cvssV3_0": {
        "score": 7.8,
        "vector": "CVSS:3.0/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H"
      },
      "cvssV3_1": {
        "score": 7.8,
        "vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H"
      },
      "cvssV4_0": {}
    },
    "nvd_repo_path": "2018/CVE-2018-10875.json",
    "references": [
      "http://lists.opensuse.org/opensuse-security-announce/2019-04/msg00021.html",
      "http://www.securitytracker.com/id/1041396",
      "https://access.redhat.com/errata/RHBA-2018:3788",
      "https://access.redhat.com/errata/RHSA-2018:2150",
      "https://access.redhat.com/errata/RHSA-2018:2151",
      "https://access.redhat.com/errata/RHSA-2018:2152",
      "https://access.redhat.com/errata/RHSA-2018:2166",
      "https://access.redhat.com/errata/RHSA-2018:2321",
      "https://access.redhat.com/errata/RHSA-2018:2585",
      "https://access.redhat.com/errata/RHSA-2019:0054",
      "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-10875",
      "https://lists.debian.org/debian-lts-announce/2019/09/msg00016.html",
      "https://usn.ubuntu.com/4072-1/",
      "https://www.debian.org/security/2019/dsa-4396"
    ],
    "title": null,
    "updated": "2021-08-04T17:14:46.777000+00:00",
    "vendors": [
      "canonical",
      "canonical$PRODUCT$ubuntu_linux",
      "debian",
      "debian$PRODUCT$debian_linux",
      "redhat",
      "redhat$PRODUCT$ansible_engine",
      "redhat$PRODUCT$ceph_storage",
      "redhat$PRODUCT$gluster_storage",
      "redhat$PRODUCT$openshift",
      "redhat$PRODUCT$openstack",
      "redhat$PRODUCT$virtualization",
      "redhat$PRODUCT$virtualization_host",
      "suse",
      "suse$PRODUCT$package_hub",
      "suse$PRODUCT$suse_linux_enterprise_server"
    ],
    "weaknesses": [
      "CWE-426"
    ]
  },
  "opencve": {
    "changes": [],
    "cpes": {
      "data": [
        "cpe:/a:redhat:ansible_engine:2.4::el7",
        "cpe:/a:redhat:ansible_engine:2.5::el7",
        "cpe:/a:redhat:ansible_engine:2.6::el7",
        "cpe:/a:redhat:ansible_engine:2::el7",
        "cpe:/a:redhat:openstack:10::el7",
        "cpe:/a:redhat:openstack:12::el7",
        "cpe:/a:redhat:openstack:13::el7",
        "cpe:/o:redhat:enterprise_linux:7::hypervisor",
        "cpe:2.3:a:redhat:ansible_engine:2.0:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:ansible_engine:2.4:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:ansible_engine:2.5:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:ansible_engine:2.6:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:ceph_storage:2.0:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:ceph_storage:3.0:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:gluster_storage:3.0.0:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:openshift:3.0:*:*:*:enterprise:*:*:*",
        "cpe:2.3:a:redhat:openstack:10:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:openstack:12:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:openstack:13:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:virtualization:4.0:*:*:*:*:*:*:*",
        "cpe:2.3:a:redhat:virtualization_host:4.0:*:*:*:*:*:*:*",
        "cpe:2.3:a:suse:package_hub:-:*:*:*:*:*:*:*",
        "cpe:2.3:o:canonical:ubuntu_linux:16.04:*:*:*:lts:*:*:*",
        "cpe:2.3:o:canonical:ubuntu_linux:18.04:*:*:*:lts:*:*:*",
        "cpe:2.3:o:canonical:ubuntu_linux:19.04:*:*:*:*:*:*:*",
        "cpe:2.3:o:debian:debian_linux:8.0:*:*:*:*:*:*:*",
        "cpe:2.3:o:debian:debian_linux:9.0:*:*:*:*:*:*:*",
        "cpe:2.3:o:suse:suse_linux_enterprise_server:12:*:*:*:*:*:*:*"
      ],
      "providers": [
        "nvd",
        "redhat"
      ]
    },
    "created": {
      "data": "2018-06-29T00:00:00+00:00",
      "provider": "redhat"
    },
    "description": {
      "data": "A flaw was found in ansible. ansible.cfg is read from the current working directory which can be altered to make it point to a plugin or a module path under the control of an attacker, thus allowing the attacker to execute arbitrary code.",
      "provider": "mitre"
    },
    "metrics": {
      "cvssV2_0": {
        "data": {
          "score": 4.6,
          "vector": "AV:L/AC:L/Au:N/C:P/I:P/A:P"
        },
        "provider": "nvd"
      },
      "cvssV3_0": {
        "data": {
          "score": 7.8,
          "vector": "CVSS:3.0/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H"
        },
        "provider": "mitre"
      },
      "cvssV3_1": {
        "data": {
          "score": 7.8,
          "vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H"
        },
        "provider": "nvd"
      },
      "cvssV4_0": {
        "data": {},
        "provider": null
      },
      "kev": {
        "data": {},
        "provider": null
      },
      "ssvc": {
        "data": {},
        "provider": null
      },
      "threat_severity": {
        "data": "Moderate",
        "provider": "redhat"
      }
    },
    "references": {
      "data": [
        "http://lists.opensuse.org/opensuse-security-announce/2019-04/msg00021.html",
        "http://www.securitytracker.com/id/1041396",
        "https://access.redhat.com/errata/RHBA-2018:3788",
        "https://access.redhat.com/errata/RHSA-2018:2150",
        "https://access.redhat.com/errata/RHSA-2018:2151",
        "https://access.redhat.com/errata/RHSA-2018:2152",
        "https://access.redhat.com/errata/RHSA-2018:2166",
        "https://access.redhat.com/errata/RHSA-2018:2321",
        "https://access.redhat.com/errata/RHSA-2018:2585",
        "https://access.redhat.com/errata/RHSA-2019:0054",
        "https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-10875",
        "https://lists.debian.org/debian-lts-announce/2019/09/msg00016.html",
        "https://nvd.nist.gov/vuln/detail/CVE-2018-10875",
        "https://usn.ubuntu.com/4072-1/",
        "https://www.cve.org/CVERecord?id=CVE-2018-10875",
        "https://www.debian.org/security/2019/dsa-4396"
      ],
      "providers": [
        "mitre",
        "nvd",
        "redhat"
      ]
    },
    "title": {
      "data": "ansible: ansible.cfg is being read from current working directory allowing possible code execution",
      "provider": "nvd"
    },
    "updated": {
      "data": "2024-08-05T07:46:47.518000+00:00",
      "provider": "mitre"
    },
    "vendors": {
      "data": [
        "canonical",
        "canonical$PRODUCT$ubuntu_linux",
        "debian",
        "debian$PRODUCT$debian_linux",
        "redhat",
        "redhat$PRODUCT$ansible_engine",
        "redhat$PRODUCT$ceph_storage",
        "redhat$PRODUCT$enterprise_linux",
        "redhat$PRODUCT$gluster_storage",
        "redhat$PRODUCT$openshift",
        "redhat$PRODUCT$openstack",
        "redhat$PRODUCT$virtualization",
        "redhat$PRODUCT$virtualization_host",
        "suse",
        "suse$PRODUCT$package_hub",
        "suse$PRODUCT$suse_linux_enterprise_server"
      ],
      "providers": [
        "nvd",
        "redhat"
      ]
    },
    "weaknesses": {
      "data": [
        "CWE-426"
      ],
      "providers": [
        "mitre",
        "nvd",
        "redhat"
      ]
    }
  },
  "redhat": {
    "cpes": [
      "cpe:/a:redhat:ansible_engine:2.4::el7",
      "cpe:/a:redhat:ansible_engine:2.5::el7",
      "cpe:/a:redhat:ansible_engine:2.6::el7",
      "cpe:/a:redhat:ansible_engine:2::el7",
      "cpe:/a:redhat:openstack:10::el7",
      "cpe:/a:redhat:openstack:12::el7",
      "cpe:/a:redhat:openstack:13::el7",
      "cpe:/o:redhat:enterprise_linux:7::hypervisor"
    ],
    "created": "2018-06-29T00:00:00+00:00",
    "description": "A flaw was found in ansible. ansible.cfg is read from the current working directory which can be altered to make it point to a plugin or a module path under the control of an attacker, thus allowing the attacker to execute arbitrary code.",
    "metrics": {
      "cvssV2_0": {},
      "cvssV3_0": {
        "score": 7.8,
        "vector": "CVSS:3.0/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H"
      },
      "cvssV3_1": {},
      "threat_severity": "Moderate"
    },
    "redhat_repo_path": "2018/CVE-2018-10875.json",
    "references": [
      "https://nvd.nist.gov/vuln/detail/CVE-2018-10875",
      "https://www.cve.org/CVERecord?id=CVE-2018-10875"
    ],
    "title": "ansible: ansible.cfg is being read from current working directory allowing possible code execution",
    "updated": "2018-06-29T00:00:00+00:00",
    "vendors": [
      "redhat",
      "redhat$PRODUCT$ansible_engine",
      "redhat$PRODUCT$enterprise_linux",
      "redhat$PRODUCT$openstack"
    ],
    "weaknesses": [
      "CWE-426"
    ]
  }
}